### PR TITLE
Refactor Code to Obey SOLID principles

### PR DESCRIPTION
**Pull request by StarMindz**
I refactored my code to obey SOLID principles
- I create two new classes Create and List to handle creating and listing out items
-  Remove all unrelated functions from the App class
-  The App class now solely handles the logic relating to running the app
- The main class is solely tasked with running the app

### CODE REVIEW

**Review by Nemwel-Boniface — CHANGES_REQUESTED** [review]
## Hello team :wave: ,
I am Nemwel your reviewer for this PR :man_beard: 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## :heavy_multiplication_x:  Invalid Code Review Request :heavy_multiplication_x: 

You have submitted a project, that does not have passing linter checks. Kindy resubmit your project after implementing this.

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me **@Nemwel-Boniface** in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

File: create_items.rb [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+    select_person = gets.chomp.to_i
+    case select_person
+    when 1
+      create_student
+    when 2
+      create_teacher
+    else
+      puts 'invalid input. Input either 1 or 2 according to your choice'
+    end
+  end
```
Comment (rloterh) [inline create_items.rb]: - [ ] A number of your methods (all create methods defined in the _create_items_ file) do not adhere to the single responsibility rule. Kindly ensure that *all* requests for input and input validation are placed in the `main.rb` file and/or classes/methods whose responsibilities are to request or validate input.` I urge you to kindly read further on the same via [this link](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/lessons/solid.md#best-practices-and-solid)

File: list_items.rb [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+    print 'Person id: '
+    person_id = gets.chomp.to_i
+
+    @rentals.each do |rental|
+      if rental.person.id.to_i == person_id
+        puts "Date: #{rental.date}, Book: '#{rental.book.title}' by #{rental.book.author}"
+      end
+    end
+  end
```
Comment (rloterh) [inline list_items.rb]: Similarly, let's ensure that our methods follow the single responsibility principle by placing all requests for input and input validation in the `main.rb` file and/or classes/methods whose responsibilities are to request or validate input. Kindly read further on the same via [this link](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/lessons/solid.md#best-practices-and-solid)

**Review by rloterh — CHANGES_REQUESTED** [review]
Hi Stanley & Marcos, 

You did extremely great on this project! 👏 👏 
There are however a few issues that you still need to work on to go to the next project but you are almost there! 🚦🚦   🏃 🏃‍♂️  

## Highlights 👍  
- Good job refactoring to improve your code so far! 

## Required Changes ♻️
_Kindly check the inline comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

- [ ] OPTIONAL
It's best practice to mark your object with _accessor attributes_ before initializing them in a constructor.

Cheers and Happy coding!👏 💻 🍷

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: list_items.rb (hunk 2 above)
Comment (StarMindz) [reply list_items.rb]: @rloterh Thank you for the review. I've implemented the changes you requested for. All requests for inputs and input validation are now in a separate class called Reader. The create_item class and list_item class now have obey Single Responsibility principle

**Review by TedLivist — APPROVED** [review]
## STATUS: APPROVED ✔️ ✔️

Hi @StarMindz and @Goruchie,

✔️ Use of SOLID principles

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. And please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**